### ISSUE TITLE
Support WASM by making file resolution a Cargo feature

### ISSUE THREAD

**Issue by bnjt** [santhosh-tekuri/boon#5]
@santhosh-tekuri firstly, thank you building this library.

We're interested in use this library in an application that targets WASM. Some URL/URI library functionality that boon depends on, such as converting a URI to a file path, isn't supported and isn't really applicable under WASM. 

Gating boon's functionality to resolve files (and possibly URLs) behind a Cargo feature flag(s), would likely be an easy-ish way to support WASM.

**Comment by santhosh-tekuri** [comment]
do you mean, you cannot compile this library to WASM target?

**Comment by bnjt** [comment]
> do you mean, you cannot compile this library to WASM target?

Correct. The `Url` library does not expose the `from_file_path` method if the target is WASM in the browser. It's gated behind the following conditional compilation config:
`#[cfg(any(unix, windows, target_os = "redox", target_os = "wasi"))]`

To be more specific, the problem is really just WASM targets like browsers that don't support WASI (e.g. things like file operations).

**Comment by santhosh-tekuri** [comment]
@bnjt I have made the fix in `wasm` branch: e8727a1

could you please test the fix from your end. that will help me in publishing to crates.io

**Comment by bnjt** [comment]
@santhosh-tekuri thank you. There's still a reference to `from_file_path` in util.rs
```
error[E0599]: no function or associated item named `from_file_path` found for struct `Url` in the current scope
  --> /Users/../Repos/boon/src/util.rs:28:21
   |
28 |         return Url::from_file_path(s)
   |                     ^^^^^^^^^^^^^^ function or associated item not found in `Url`
```

 I've proposed a slightly different fix too in PR #6. It exposes new methods

**Comment by bnjt** [comment]
I've proposed a slightly different fix too in PR #6. It introduces a Cargo feature `resolve-file` and new methods to be used when it is not enabled, which provides some level of static indication that the user should not pass in a string that represents file path.

**Comment by santhosh-tekuri** [comment]
@bnjt got it. I added cfg for that line. please recheck

**Comment by santhosh-tekuri** [comment]
I looked #6. WASM developers already know that they can't use files. then why confuse non-wasm user by having two methods one with file and one without file.

**Comment by bnjt** [comment]
Updated fix compiles under WASM. Thanks for the quick response!

> then why confuse non-wasm user by having two methods one with file and one without file.
It's more confusing for it to fail at runtime. Think of the developer who didn't add the dependency to their project and isn't aware of this limitation. The extra cargo feature flag and typed methods provide static enforcement that the location must be a `Url` type and not file type.

**Comment by santhosh-tekuri** [comment]
published in v0.5.1
